activerecord/test/support/schema_dumping_helper.rb

```diff
@@ -11,7 +11,7 @@ def dump_table_schema(table, connection = ActiveRecord::Base.connection)
     ActiveRecord::SchemaDumper.ignore_tables = old_ignore_tables
   end
 
-  def dump_all_table_schema(ignore_tables)
+  def dump_all_table_schema(ignore_tables = [])
```

Comment:
With this change, we do not have to have a dedicated `perform_schema_dump` method in activerecord/test/cases/schema_dumper_test.rb . What do you think about changing them as follows?

```diff
% git diff
diff --git a/activerecord/test/cases/schema_dumper_test.rb b/activerecord/test/cases/schema_dumper_test.rb
index 336f7c6dda..964f50940f 100644
--- a/activerecord/test/cases/schema_dumper_test.rb
+++ b/activerecord/test/cases/schema_dumper_test.rb
@@ -12,11 +12,7 @@ class SchemaDumperTest < ActiveRecord::TestCase
   end

   def standard_dump
-    @@standard_dump ||= perform_schema_dump
-  end
-
-  def perform_schema_dump
-    dump_all_table_schema []
+    @@standard_dump ||= dump_all_table_schema
   end

   def test_dump_schema_information_with_empty_versions
@@ -354,13 +350,13 @@ def test_schema_dump_includes_extensions
       connection = ActiveRecord::Base.connection

       connection.stub(:extensions, ["hstore"]) do
-        output = perform_schema_dump
+        output = dump_all_table_schema
         assert_match "# These are extensions that must be enabled", output
         assert_match %r{enable_extension "hstore"}, output
       end

       connection.stub(:extensions, []) do
-        output = perform_schema_dump
+        output = dump_all_table_schema
         assert_no_match "# These are extensions that must be enabled", output
         assert_no_match %r{enable_extension}, output
       end
@@ -370,13 +366,13 @@ def test_schema_dump_includes_extensions_in_alphabetic_order
       connection = ActiveRecord::Base.connection

       connection.stub(:extensions, ["hstore", "uuid-ossp", "xml2"]) do
-        output = perform_schema_dump
+        output = dump_all_table_schema
         enabled_extensions = output.scan(%r{enable_extension "(.+)"}).flatten
         assert_equal ["hstore", "uuid-ossp", "xml2"], enabled_extensions
       end

       connection.stub(:extensions, ["uuid-ossp", "xml2", "hstore"]) do
-        output = perform_schema_dump
+        output = dump_all_table_schema
         enabled_extensions = output.scan(%r{enable_extension "(.+)"}).flatten
         assert_equal ["hstore", "uuid-ossp", "xml2"], enabled_extensions
       end
@@ -427,7 +423,7 @@ def test_do_not_dump_foreign_keys_when_bypassed_by_config
           }
         )

-        output = perform_schema_dump
+        output = dump_all_table_schema
         assert_no_match(/^\s+add_foreign_key "fk_test_has_fk"[^\n]+\n\s+add_foreign_key "lessons_students"/, output)
       ensure
         ActiveRecord::Base.establish_connection(:arunit)
@@ -461,7 +457,7 @@ def test_schema_dump_with_table_name_prefix_and_suffix
     migration = CreateDogMigration.new
     migration.migrate(:up)

-    output = perform_schema_dump
+    output = dump_all_table_schema
     assert_no_match %r{create_table "foo_.+_bar"}, output
     assert_no_match %r{add_index "foo_.+_bar"}, output
     assert_no_match %r{create_table "schema_migrations"}, output
@@ -486,7 +482,7 @@ def test_schema_dump_with_table_name_prefix_and_suffix_regexp_escape
     migration = CreateDogMigration.new
     migration.migrate(:up)

-    output = perform_schema_dump
+    output = dump_all_table_schema
     assert_no_match %r{create_table "foo\$.+\$bar"}, output
     assert_no_match %r{add_index "foo\$.+\$bar"}, output
     assert_no_match %r{create_table "schema_migrations"}, output
@@ -554,7 +550,7 @@ def down
       end
       migration.migrate(:up)

-      output = perform_schema_dump
+      output = dump_all_table_schema
       assert output.include?('t.datetime "this_should_remain_datetime"')
       assert output.include?('t.datetime "this_is_an_alias_of_datetime"')
       assert output.include?('t.datetime "without_time_zone"')
@@ -583,7 +579,7 @@ def down
         end
         migration.migrate(:up)

-        output = perform_schema_dump
+        output = dump_all_table_schema
         assert output.include?('t.datetime "this_should_remain_datetime"')
         assert output.include?('t.datetime "this_is_an_alias_of_datetime"')
         assert output.include?('t.timestamp "without_time_zone"')
@@ -611,7 +607,7 @@ def down
       end
       migration.migrate(:up)

-      output = perform_schema_dump
+      output = dump_all_table_schema
       assert output.include?('t.datetime "this_should_remain_datetime"')
       assert output.include?('t.datetime "this_is_an_alias_of_datetime"')
       assert output.include?('t.datetime "this_is_also_an_alias_of_datetime"')
@@ -638,7 +634,7 @@ def down
         end
         migration.migrate(:up)

-        output = perform_schema_dump
+        output = dump_all_table_schema
         # Normally we'd write `t.datetime` here. But because you've changed the `datetime_type`
         # to something else, `t.datetime` now means `:timestamptz`. To ensure that old columns
         # are still created as a `:timestamp` we need to change what is written to the schema dump.
@@ -672,7 +668,7 @@ def down
       end
       migration.migrate(:up)

-      output = perform_schema_dump
+      output = dump_all_table_schema
       assert output.include?('t.datetime "default_format"')
       assert output.include?('t.datetime "without_time_zone"')
       assert output.include?('t.timestamptz "with_time_zone"')
@@ -680,7 +676,7 @@ def down
       datetime_type_was = ActiveRecord::ConnectionAdapters::PostgreSQLAdapter.datetime_type
       ActiveRecord::ConnectionAdapters::PostgreSQLAdapter.datetime_type = :timestamptz

-      output = perform_schema_dump
+      output = dump_all_table_schema
       assert output.include?('t.timestamp "default_format"')
       assert output.include?('t.timestamp "without_time_zone"')
       assert output.include?('t.datetime "with_time_zone"')
@@ -708,7 +704,7 @@ def down
       end
       migration.migrate(:up)

-      output = perform_schema_dump
+      output = dump_all_table_schema
       assert output.include?('t.datetime "default_format"')
       assert output.include?('t.datetime "without_time_zone"')
       assert output.include?('t.datetime "also_without_time_zone"')
@@ -736,7 +732,7 @@ def down
       end
       migration.migrate(:up)

-      output = perform_schema_dump
+      output = dump_all_table_schema
       assert output.include?('t.datetime "default_format"')
       assert output.include?('t.datetime "without_time_zone"')
       assert output.include?('t.datetime "also_without_time_zone"')
@@ -763,7 +759,7 @@ def down
       end
       migration.migrate(:up)

-      output = perform_schema_dump
+      output = dump_all_table_schema
       assert output.include?('t.datetime "default_format"')
       assert output.include?('t.datetime "without_time_zone"')
       assert output.include?('t.datetime "also_without_time_zone"')
@@ -789,7 +785,7 @@ def down
         end
         migration.migrate(:up)

-        output = perform_schema_dump
+        output = dump_all_table_schema
         # Normally we'd write `t.datetime` here. But because you've changed the `datetime_type`
         # to something else, `t.datetime` now means `:timestamptz`. To ensure that old columns
         # are still created as a `:timestamp` we need to change what is written to the schema dump.
@@ -822,7 +818,7 @@ def down
         end
         migration.migrate(:up)

-        output = perform_schema_dump
+        output = dump_all_table_schema
         # Normally we'd write `t.datetime` here. But because you've changed the `datetime_type`
         # to something else, `t.datetime` now means `:timestamptz`. To ensure that old columns
         # are still created as a `:timestamp` we need to change what is written to the schema dump.
%
```

Reply:
It is up to you if you accept this change. Also, you can open a separate pull request once this pull request is merged because changing activerecord/test/cases/schema_dumper_test.rb looks like another topic.

Reply:
Good idea. I'll include this.